cobbler/cobbler_collections/images.py

```diff
@@ -43,34 +43,38 @@ def factory_produce(self, api: "CobblerAPI", seed_data: Dict[str, Any]):
 
     def remove(
         self,
-        name: str,
+        ref: image.Image,
         with_delete: bool = True,
         with_sync: bool = True,
         with_triggers: bool = True,
         recursive: bool = True,
         rebuild_menu: bool = True,
     ) -> None:
         """
-        Remove element named 'name' from the collection
-
-        :raises CX: In case object does not exist or it would orhan a system.
+        Remove the given element from the collection
+
+        :param ref: The object to delete
+        :param with_delete: In case the deletion triggers are executed for this image.
+        :param with_sync: In case a Cobbler Sync should be executed after the action.
+        :param with_triggers: In case the Cobbler Trigger mechanism should be executed.
+        :param recursive: In case you want to delete all objects this image references.
+        :param rebuild_menu: unused
+        :raises CX: Raised in case you want to delete a none existing image.
         """
         # rebuild_menu is not used
         _ = rebuild_menu
-        # NOTE: with_delete isn't currently meaningful for repos but is left in for consistency in the API. Unused.
-        obj = self.listing.get(name, None)
 
-        if obj is None:
-            raise CX(f"cannot delete an object that does not exist: {name}")
+        if ref is None:  # type: ignore
+            raise CX("cannot delete an object that does not exist")
 
         # first see if any Groups use this distro
         if not recursive:
             for system in self.api.systems():
```

Comment:
I suggest here, as in `Distros`, to use `api.find` instead of a loop. There can be a lot of systems, so searching through the index will most likely be more effective.
```python
kids = self.api.find_system(return_list=True, image=ref.uid)
```